v-next/hardhat-chai-matchers/test/helpers/pretest.ts

```diff
@@ -0,0 +1,28 @@
+import path, { dirname } from "node:path";
+import { fileURLToPath } from "node:url";
+
+import { createHardhatRuntimeEnvironment } from "@ignored/hardhat-vnext/hre";
+
+// Compile the contracts before executing the tests to bypass the compilation step during testing.
+(async () => {
```

Comment:
This is not needed. In ESM you can use top-level await.

If the linter was complaining, let's add an exception with an explanation instead.

Reply:
Yes, the linter is complaining. I added an exception. See [here](https://github.com/NomicFoundation/hardhat/pull/5946/commits/237ddc4ba08311b968565a391449687d34496eac)